.CMake/compiler_opts.cmake

```diff
@@ -108,6 +108,7 @@ if(CMAKE_C_COMPILER_ID MATCHES "Clang")
         set(THREADS_PREFER_PTHREAD_FLAG ON)
         find_package(Threads REQUIRED)
         set(OQS_USE_PTHREADS_IN_TESTS 1)
+        set(OQS_USE_PTHREADS_IN_LIBOQS 1)
```

Comment:
Are you 100% sure pthreads is available if `clang` is the compiler?

Reply:
I am using the same logic as test_kem and test_sig for [`#include`ing `<pthread.h>`](https://github.com/open-quantum-safe/liboqs/blob/a80fdf8ee48ea979b5b90ceba026610338608ce0/tests/test_kem.c#L10) and [linking the library](https://github.com/open-quantum-safe/liboqs/blob/a80fdf8ee48ea979b5b90ceba026610338608ce0/tests/CMakeLists.txt#L72), so I am perplexed about why it's not working here. Besides, the `cmake` step should fail if pthreads is not found since the `find_package` is called with `REQUIRED`.

Reply:
> I am perplexed about why it's not working here

Me as well -- but there is a conceptual difference in this PR: It adds the pthreads dependency to the library. In testing, it's with the executable directly (where the executable linker may resolve the dependency automatically). As to why your explicit reference to "Threads::Threads" isn't honoured, I cannot comment on. [SO comments that `-lpthread` needs to be added](https://stackoverflow.com/questions/59508259/in-depth-explanation-for-why-we-need-pthread-in-linker-option-for-gcc).

> Besides, the cmake step should fail if pthreads is not found since the find_package is called with REQUIRED.

This is exactly what I don't find very appealing: This will break builds of `liboqs` if "pthreads" is not available. Wouldn't it be better to not break (but behave as it does right now)? But then again, the current behaviour isn't right -- so if we need to add pthreads to make it operate correctly with threading, so be it. Community opinions solicited.

Reply:
> This will break builds of liboqs if "pthreads" is not available.
The `find_package(Threads REQUIRED)` call was added by @dstebila in 4cc0398. Can defer to him but I think it can be made not-required without breaking anything, and then conditioning pthread usage on `Threads_FOUND`.

The above is separate from making pthreads link, which I am still working on.